### PR TITLE
linux_6_3: drop as EOL

### PR DESCRIPTION
**Pull request by fabianhjr**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/zfs/unstable.nix [hunk 1]
```diff
@@ -11,17 +11,11 @@ let
...
-  #   zfs-2.1.9<=x<=2.1.10 is broken with aarch64-linux-6.2
-  #   for future releases, please delete this condition.
   kernelCompatible = if stdenv'.isx86_64 || removeLinuxDRM
     then kernel.kernelOlder "6.4"
     else kernel.kernelOlder "6.2";
 
-  latestCompatibleLinuxPackages = if stdenv'.isx86_64 || removeLinuxDRM then
-    linuxKernel.packages.linux_6_3
-  else
-    linuxKernel.packages.linux_6_1;
```
Comment (RaitoBezarius) [inline pkgs/os-specific/linux/zfs/unstable.nix:24]: If you can give me sometime, I will just look if ZFS works on 6_4.

Comment (RaitoBezarius) [reply pkgs/os-specific/linux/zfs/unstable.nix:24]: It works, please keep the if and just put 6_4 instead of 6_3.

Comment (fabianhjr) [reply pkgs/os-specific/linux/zfs/unstable.nix:24]: Sure, changing to draft.

Comment (fabianhjr) [reply pkgs/os-specific/linux/zfs/unstable.nix:24]: Also, no rush, I don't think dropping this kernel is time sensitive. :3

Comment (Shawn8901) [reply pkgs/os-specific/linux/zfs/unstable.nix:24]: 2.1.12 and also the 2.2.0rc2 are currently just mentioning 6.3 as highest supported, so merging that as it is would mean to downgrade zfs users if the are using `latestCompatibleLinuxPackages` as kernel package.

Comment (fabianhjr) [reply pkgs/os-specific/linux/zfs/unstable.nix:24]: will wait for 2.2.0 release

Comment (RaitoBezarius) [reply pkgs/os-specific/linux/zfs/unstable.nix:24]: @Shawn8901 2.1.12 has already 6.4 support, 2.2 has 6.5 support

Comment (RaitoBezarius) [reply pkgs/os-specific/linux/zfs/unstable.nix:24]: Ah… Indeed the META have 6.3 maximal… Surprising.

Comment (RaitoBezarius) [reply pkgs/os-specific/linux/zfs/unstable.nix:24]: Let's proceed now.

**Review by RaitoBezarius — APPROVED** [review]
Ready to go. We don't need to wait on ZFS anymore.

**Comment by fabianhjr** [comment]
To confirm, leaving the 6.3 → 6.1 change?

**Comment by RaitoBezarius** [comment]
> To confirm, leaving the 6.3 → 6.1 change?

Correct. Downgrade is fine.

**Comment by Ma27** [comment]
These are also EOL on 23.05, so we should remove them there as well.

**Comment by fabianhjr** [comment]
Hi, will make the cherry-pick manually. Thanks for the heads up will do. :3

**Comment by Ma27** [comment]
Thanks for taking care of that! Was about to do it tomorrow otherwise :)

**Pull request merged**